test/integration/pipeline/batch_pipeline_test.cpp

```diff
@@ -0,0 +1,295 @@
+                          EXPECT_NO_THROW(boost::apply_visitor(
+                              framework::SpecifiedVisitor<
+                                  interface::StatelessValidTxResponse>(),
+                              status.get()));
+                        }
+                      })
+      .checkProposal([&transaction_sequence](const auto proposal) {
+        ASSERT_THAT(
+            proposal->transactions(),
+            Pointwise(RefAndPointerEq(), transaction_sequence.transactions()));
+      })
+      .checkVerifiedProposal([](const auto verified_proposal) {
+        ASSERT_THAT(verified_proposal->transactions(), IsEmpty());
+      })
+      .done();
+}
+
+/**
+ * @given ordered batch of three transactions, with one transaction being
+ * stateful invalid
+ * @when batch is sent to iroha
+ * @then all transactions except stateful invalid one are committed
+ */
+TEST_F(BatchPipelineTest, InvalidOrderedBatch) {
+  auto batch_transactions = framework::batch::makeTestBatchTransactions(
+      interface::types::BatchType::ORDERED,
+      prepareTransferAssetBuilder(kFirstUserId, kSecondUserId, kAssetA, "0.3"),
+      prepareTransferAssetBuilder(
+          kSecondUserId,
+          kFirstUserId,
+          kAssetB,
+          "2.0"),  // stateful invalid tx due to too big transfer
+      prepareTransferAssetBuilder(kFirstUserId, kSecondUserId, kAssetA, "0.7"));
+
+  SCOPED_TRACE("From InvalidOrderedBatch");
+  auto transaction_sequence = createTransactionSequence(
+      {signedTx(batch_transactions[0], kFirstUserKeypair),
+       signedTx(batch_transactions[1], kSecondUserKeypair),
+       signedTx(batch_transactions[2], kFirstUserKeypair)});
+
+  integration_framework::IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTxAwait(createFirstUser(), [](const auto &) {})
+      .sendTxAwait(createSecondUser(), [](const auto &) {})
+      .sendTxAwait(
+          createAndAddAssets(kFirstUserId, kAssetA, "1.0", kFirstUserKeypair),
+          [](const auto &) {})
+      .sendTxAwait(
+          createAndAddAssets(kSecondUserId, kAssetB, "1.0", kSecondUserKeypair),
+          [](const auto &) {})
+      .sendTxSequenceAwait(
+          transaction_sequence,
+          [&](const auto block) {
+            ASSERT_THAT(
+                block->transactions(),
+                ElementsAre(
+                    RefAndPointerEq(transaction_sequence.transactions()[0]),
+                    RefAndPointerEq(transaction_sequence.transactions()[2])));
+          })
+      .done();
+}
+
+INSTANTIATE_TEST_CASE_P(BatchPipelineParameterizedTest,
+                        BatchPipelineTest,
+                        // note additional comma is needed to make it compile
+                        // https://github.com/google/googletest/issues/1419
+                        Values(interface::types::BatchType::ATOMIC,
+                               interface::types::BatchType::ORDERED), );
```

Comment:
Regards to https://soramitsu.atlassian.net/browse/IR-1537 task. Also, there is required to write a test with MST and signatures.
I have missed this test or there are some troubles with that?